<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>createEndpoint</class>
 <widget class="QDialog" name="createEndpoint">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>640</width>
    <height>480</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Create an endpoint</string>
  </property>
  <property name="locale">
   <locale language="English" country="UnitedStates"/>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout">
   <item>
    <layout class="QHBoxLayout" name="horizontalLayout">
     <item>
      <widget class="QLabel" name="label">
       <property name="sizePolicy">
        <sizepolicy hsizetype="Maximum" vsizetype="Preferred">
         <horstretch>0</horstretch>
         <verstretch>0</verstretch>
        </sizepolicy>
       </property>
       <property name="text">
        <string>Creating an endpoint for the</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QComboBox" name="pbRequestCombo"/>
     </item>
     <item>
      <widget class="QLabel" name="label_5">
       <property name="text">
        <string>Protobuf message</string>
       </property>
      </widget>
     </item>
    </layout>
   </item>
   <item>
    <layout class="QFormLayout" name="formLayout">
     <item row="0" column="0">
      <widget class="QLabel" name="endpointUrlLabel">
       <property name="text">
        <string>Endpoint URL:</string>
       </property>
      </widget>
     </item>
     <item row="0" column="1">
      <widget class="QLineEdit" name="endpointUrl"/>
     </item>
    </layout>
   </item>
   <item>
    <widget class="QLabel" name="label_2">
     <property name="text">
      <string>Transport type:</string>
     </property>
    </widget>
   </item>
   <item>
    <widget class="QListWidget" name="transports"/>
   </item>
   <item>
    <widget class="QWidget" name="reqDataSubform" native="true">
     <layout class="QVBoxLayout" name="reqDataSubform_2">
      <property name="leftMargin">
       <number>0</number>
      </property>
      <property name="topMargin">
       <number>0</number>
      </property>
      <property name="rightMargin">
       <number>0</number>
      </property>
      <property name="bottomMargin">
       <number>0</number>
      </property>
      <item>
       <widget class="QLabel" name="sampleDataLabel">
        <property name="text">
         <string>Sample request data, one per line (pick a transport type first):</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QPlainTextEdit" name="sampleData"/>
      </item>
      <item>
       <widget class="QWidget" name="pbParamSubform" native="true">
        <layout class="QFormLayout" name="pbParamSubform_2">
         <item row="0" column="0">
          <widget class="QLabel" name="pbParamKeyLabel">
           <property name="text">
            <string>Key of parameter containing Protobuf data in request data:</string>
           </property>
          </widget>
         </item>
         <item row="0" column="1">
          <widget class="QLineEdit" name="pbParamKey">
           <property name="sizePolicy">
            <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
             <horstretch>0</horstretch>
             <verstretch>0</verstretch>
            </sizepolicy>
           </property>
          </widget>
         </item>
        </layout>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <layout class="QHBoxLayout" name="horizontalLayout_2">
     <item>
      <widget class="QCheckBox" name="parsePbCheckbox">
       <property name="sizePolicy">
        <sizepolicy hsizetype="Maximum" vsizetype="Fixed">
         <horstretch>0</horstretch>
         <verstretch>0</verstretch>
        </sizepolicy>
       </property>
       <property name="text">
        <string>Parse response as a Protobuf message</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QComboBox" name="pbRespCombo">
       <property name="sizePolicy">
        <sizepolicy hsizetype="Expanding" vsizetype="Fixed">
         <horstretch>0</horstretch>
         <verstretch>0</verstretch>
        </sizepolicy>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QPushButton" name="loadRespPbBtn">
       <property name="text">
        <string>Load from another .proto...</string>
       </property>
      </widget>
     </item>
    </layout>
   </item>
   <item>
    <widget class="QDialogButtonBox" name="buttonBox">
     <property name="orientation">
      <enum>Qt::Horizontal</enum>
     </property>
     <property name="standardButtons">
      <set>QDialogButtonBox::Cancel|QDialogButtonBox::Ok</set>
     </property>
    </widget>
   </item>
  </layout>
 </widget>
 <resources/>
 <connections>
  <connection>
   <sender>buttonBox</sender>
   <signal>rejected()</signal>
   <receiver>createEndpoint</receiver>
   <slot>reject()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>316</x>
     <y>260</y>
    </hint>
    <hint type="destinationlabel">
     <x>286</x>
     <y>274</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
